components/builder-originsrv/src/migrations/origin_packages.rs

```diff
@@ -42,44 +42,94 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
              )"#,
     )?;
     migrator.migrate("originsrv",
-                 r#"CREATE OR REPLACE FUNCTION insert_origin_package_v1 (
-                    op_origin_id bigint,
-                    op_owner_id bigint,
-                    op_name text,
-                    op_ident text,
-                    op_checksum text,
-                    op_manifest text,
-                    op_config text,
-                    op_target text,
-                    op_deps text,
-                    op_tdeps text,
-                    op_exposes text
-                 ) RETURNS SETOF origin_packages AS $$
-                     DECLARE
-                        inserted_package origin_packages;
-                        channel_id bigint;
-                     BEGIN
-                         INSERT INTO origin_packages (origin_id, owner_id, name, ident, checksum, manifest, config, target, deps, tdeps, exposes)
-                                VALUES (op_origin_id, op_owner_id, op_name, op_ident, op_checksum, op_manifest, op_config, op_target, op_deps, op_tdeps, op_exposes)
-                                RETURNING * into inserted_package;
+                 r#"CREATE OR REPLACE FUNCTION insert_origin_package_v2 (
```

Comment:
These are great questions that, in combination with @raskchanky's `hab pkg channels` PR (#2704) and a conversation with him about it today, have led me reconsider whether this change the right one for us right now.

Since all the UI needs is to display a list of channels by package, an API just for that purpose makes sense. I'm going to close this PR, rebase my UI changes onto #2704, and submit that; which bring package channels into the package-detail view. Then @raskchanky and I will collaborate on a separate change that'll use his new API to pull channels into the other APIs that need them.